packages/components/ng-ovh-uirouter-layout/src/index.js

```diff
@@ -0,0 +1,205 @@
+            });
+          }
+          // if backdrop is clicked - be sure to close the modal
+          modalInstance.result.catch(() => $state.go(get(state, 'layout.redirectTo')));
+        }
+      });
+    });
+
+    /* -----  End of UI ROUTER MODAL LAYOUT DEFINITION  ------ */
+  })
+  .run(/* @ngInject */($transitions, $state, $stateRegistry, $urlService) => {
+    /* ========================================================
+    =            UI ROUTER MODAL LAYOUT DEFINITION            =
+    ========================================================= */
+
+    /**
+     * As initial URL synchronization is delayed we can check all modal layout states and check
+     * if toChilds attribute is setted.
+     * For these states we will need to create new states as follow:
+     * We will take the direct parent of the modal layout state and create new states with the same
+     * layout configuration to all of its child states.
+     */
+    const layoutStates = filter($stateRegistry.states, ({ layout }) => get(layout, 'toChilds') === true || size(get(layout, 'toChilds', [])));
+    const getChildStates = parentStateName => filter($stateRegistry.states, ({ name }) => startsWith(name, `${parentStateName}.`));
+
+    layoutStates.forEach((layoutState) => {
+      let childStates;
+
+      // build child states that need modal layout applied
+      if (angular.isArray(layoutState.layout.toChilds)) {
+        childStates = layoutState.layout.toChilds;
+      } else {
+        childStates = map(getChildStates(layoutState.parent.name), 'name');
+      }
+
+      // build child states that need to be ignored
+      // 1st: all child states of each states that need to be ignored
+      // 2nd: current layout state doesn't need to have itself as modal child
+      layoutState.layout.ignoreChilds.forEach((childState) => {
+        set(layoutState, 'layout.ignoreChilds', layoutState.layout.ignoreChilds.concat(map(getChildStates(childState), 'name')));
+      });
+      layoutState.layout.ignoreChilds.push(layoutState.name);
+
+      // remove child states that need to be ignored
+      childStates = xor(childStates, layoutState.layout.ignoreChilds);
+
+      // create child state with layout settings applied
+      const modalSateSuffix = last(layoutState.name.split('.'));
+      childStates.forEach((childState) => {
+        $stateRegistry.register({
+          name: `${childState}.${modalSateSuffix}`,
+          url: layoutState.self.url,
+          templateUrl: layoutState.self.templateUrl,
+          controller: layoutState.self.controller,
+          component: layoutState.self.component,
+          layout: { // don't know why full config must be defined???
+            name: 'modal',
+            toChilds: false,
+            ignoreChilds: [],
+          },
+        });
+      });
+    });
+
+    $urlService.listen(); // Start responding to URL changes
```

Comment:
I'm not sure we need to stop and listen URL changes, because modules which are defining the routing will require this module :)

Cf: https://docs.angularjs.org/guide/module

> After that, the configuration blocks (config functions) are executed. This means the configuration blocks of the required modules execute before the configuration blocks of any requiring module.

We should remove this I think, because config for state will always be executed after config of this module (if dependency are proper) :)

Reply:
Removed.

Reply:
nice ;)